app/models/container_deployment.rb

```diff
@@ -0,0 +1,11 @@
+class ContainerDeployment < ApplicationRecord
+  has_many :container_deployment_authentications
+  belongs_to :deployed_ems, :class_name => 'ExtManagementSystem'
+  belongs_to :deployed_on_ems, :class_name => 'ExtManagementSystem'
```

Comment:
also the naming of both is not very intention revealing to me.
Is `deployed_ems` the ems that performed the deployement and the `deployed_on_ems` the one on which it was performed? 
Maybe you can explain what these should mean and a native speaker can come up with a better naming.